test/functional/unified-spec-runner/operations.ts

```diff
@@ -294,19 +324,36 @@ operations.set('insertMany', async ({ entities, operation }) => {
 });
 
 operations.set('iterateUntilDocumentOrError', async ({ entities, operation }) => {
-  const changeStream = entities.getEntity('stream', operation.object);
-  // Either change or error promise will finish
-  return Promise.race([
-    changeStream.eventCollector.waitAndShiftEvent('change'),
-    changeStream.eventCollector.waitAndShiftEvent('error')
-  ]);
+  try {
+    const changeStream = entities.getEntity('stream', operation.object);
+    // Either change or error promise will finish
+    return Promise.race([
+      changeStream.eventCollector.waitAndShiftEvent('change'),
+      changeStream.eventCollector.waitAndShiftEvent('error')
+    ]);
+  } catch (e) {
+    const findCursor = entities.getEntity('cursor', operation.object);
+    return await findCursor.next();
+  }
+});
+
+operations.set('listCollections', async ({ entities, operation }) => {
+  const db = entities.getEntity('db', operation.object);
+  const { filter, batchSize } = operation.arguments;
+  return await db.listCollections(filter, { batchSize: batchSize }).toArray();
```

Comment:
nit: drop await here

Reply:
Removed it here and for `listIndexes` as well. I was hoping that was the cause of the mystery hang but alas, it was not.